contracts/PetAdoption.sol

```diff
@@ -1,42 +1,103 @@
 // SPDX-License-Identifier: UNLICENSED
-pragma solidity ^0.8.9;
-// import "hardhat/console.sol";
+pragma solidity ^0.8.27;
 
 contract PetAdoption{
-    address public owner;
-    uint public petIndex = 0;
-    uint[] public allAdoptedPets;
+    // Struct to store pet details
+    struct Pet {
+        uint256 id;
+        string category;
+        string breed;
+    }
+    address private immutable i_owner;
```

Comment:
why you are making address private?

Reply:
We already have a function to get the address of owner, as such there is no requirement to leave it public. Also private and immutable are a little more gas efficient.

Reply:
We don't want the owner to be private.